### PR TITLE
Add TargetStrip to msg.Service

### PR DESCRIPTION
**Pull request by miekg**
From the README.md: 

If you ask for a service who's Host value is an IP address you would (in theory) get 
 back a SRV record such as: 
 
     % dig @localhost 4.rails.staging.east.skydns.local SRV
 
     ;; ANSWER SECTION:
     4.rails.staging.east.skydns.local 3600 IN SRV 10 100 8080 10.0.1.125
 
 Where the target of the SRV is an IP address. This is not how SRV records work.
 SkyDNS will in this case synthesize a domain name and add the actual IP
 address to the additional section of the response:
 
     % dig @localhost 4.rails.staging.east.skydns.local SRV
 
     ;; ANSWER SECTION:
     4.rails.staging.east.skydns.local 3600 IN SRV 10 100 4.rails.staging.east.skydns.local.
 
     ;; ADDITIONAL SECTION:
     4.rails.staging.east.skydns.local. 3600 IN A    10.0.1.125
 
 Which conveys the same information and is legal in the DNS. To have some control on how 
 the target names look you can register a service with `TargetStrip` set to a none-zero
 value. This tells SkyDNS to strip `TargetStrip` labels from the left-side of the generated
 Target domain name. Support `TargetStrip` is 2 for this service, to above response would be: 
 
     ;; ANSWER SECTION:
     4.rails.staging.east.skydns.local 3600 IN SRV 10 100 staging.east.skydns.local.
 
     ;; ADDITIONAL SECTION:
     staging.east.skydns.local. 3600 IN A    10.0.1.125

### CODE REVIEW

**Comment by miekg** [comment]
@bketelsen this also cleans up the README quite a bit and I jump the version, so presumably we should re-tag for the docker image?

And can you look it over a bit, just to make sure I'm not loosing my mind here? :-)

**Comment by bketelsen** [comment]
i want to change the docker image to one that is much smaller.  I'll try to do that this morning, after I look at these changes.

File: README.md [hunk 1]
```diff
@@ -264,16 +292,46 @@ Get all Services in staging.east:
...
+    % dig @localhost 4.rails.staging.east.skydns.local SRV
+
+    ;; ANSWER SECTION:
+    4.rails.staging.east.skydns.local 3600 IN SRV 10 100 4.rails.staging.east.skydns.local.
+
+    ;; ADDITIONAL SECTION:
+    4.rails.staging.east.skydns.local. 3600 IN A    10.0.1.125
+
+Which conveys the same information and is legal in the DNS. To have some control on how
+the target names look you can register a service with `TargetStrip` set to a none-zero
```
Comment (bketelsen) [inline README.md]: non-zero

File: README.md [hunk 2]
```diff
@@ -264,16 +292,46 @@ Get all Services in staging.east:
...
+value. This tells SkyDNS to strip `TargetStrip` labels from the left-side of the generated
+Target domain name. Support `TargetStrip` is 2 for this service, to above response would be:
+
```
Comment (bketelsen) [inline README.md]: Not clear what TargetStrip does - suggest:
Setting TargetStrip to "2" strips 2 labels from the generated target name.  (if that is what it does)

File: README.md [hunk 3]
```diff
@@ -431,43 +509,92 @@ In the example here, we don't use an UUID, we use `public.addresses`:
...
+ports on bar.skydns.local and this name has IP addres 192.168.0.1:
+
+    ;; ANSWER SECTION:
+    bar.skydns.local.   3600    IN  SRV 10 50 80 bar.skydns.local.
+    bar.skydns.local.   3600    IN  SRV 10 50 443 bar.skydns.local.
+
+    ;; ADDITIONAL SECTION:
+    bar.skydns.local. 3600    IN  A   192.168.0.1
+
+So you register:
```
Comment (bketelsen) [inline README.md]: So you register a "dummy" host named `x1`.

File: server/server.go [hunk 4]
```diff
@@ -465,29 +437,52 @@ func (s *server) AddressRecords(q dns.Question, name string, previousRecords []d
 		ip := net.ParseIP(serv.Host)
 		switch {
 		case ip == nil:
-			// TODO: deduplicate with above code
-			// Try to resolve as CNAME if it's not an IP.
+			// Try to resolve as CNAME if it's not an IP, but only is we don't
```
Comment (bketelsen) [inline server/server.go]: only if

**Comment by bketelsen** [comment]
LGTM modulo minor nits

File: README.md (hunk 2 above)
Comment (miekg) [inline README.md]: [ Quoting <notifications@github.com> in "Re: [skydns] Add TargetStrip to msg..." ]
>> +the target names look you can register a service with `TargetStrip` set to a none-zero
>> +value. This tells SkyDNS to strip `TargetStrip` labels from the left-side of the generated
>> +Target domain name. Support `TargetStrip` is 2 for this service, to above response would be:
>> +
>
>Not clear what TargetStrip does - suggest:
>Setting TargetStrip to "2" strips 2 labels from the generated target name.  (if that is what it does)

Ack. Fix the paragraph a bit as well.

**Comment by miekg** [comment]
awesome and thank you.

**Pull request merged**